lib/cdo/graphics/certificate_image.rb

```diff
@@ -15,31 +19,68 @@ def create_certificate_image2(image_path, name, params={})
   pointsize = 68
   x_offset = params[:x] || 0
   y_offset = params[:y] || 0
-  apply_text(background, name, pointsize, font, color, x_offset, y_offset)
+
+  apply_text(background, name, pointsize, font, color, x_offset, y_offset, CERT_NAME_AREA_WIDTH, CERT_NAME_AREA_HEIGHT)
   background
 end
 
-# applies the given text to the given image object.
-def apply_text(image, text, pointsize, font, color, x_offset, y_offset)
+# Applies the given text to the given image object.
+# The text is shrunk to fit the given width and height.
+#
+# @param [Magick::Image] image is the background to put text on.
+# @param [String] text is the string to add to the image.
+# @param [Integer] pointsize is the max size to display the text at.
+# @param [String] font to use e.g. 'Helvetica bold'
+# @param [String] color of the text e.g. 'rgb(118,101,160)'
+# @param [Integer] x_offset of the position to center the text at.
+# @param [Integer] y_offset of the position to center the text at.
+# @param [Integer] width in pixels of the bounding box for the text.
+# @param [Integer] height in pixels of the bounding box for the text.
+def apply_text(image, text, pointsize, font, color, x_offset, y_offset, width=nil, height=nil)
```

Comment:
What happens if width and/or height are not provided?  Also should the comment imply they are optional?